packages/monaco/src/browser/monaco-snippet-suggest-provider.ts

```diff
@@ -24,17 +24,66 @@ export class MonacoSnippetSuggestProvider implements monaco.modes.ISuggestSuppor
     @inject(FileSystem)
     protected readonly filesystem: FileSystem;
 
-    protected readonly snippets = new Map<string, MonacoSnippetSuggestion[]>();
+    protected readonly snippets = new Map<string, Snippet[]>();
     protected readonly pendingSnippets = new Map<string, Promise<void>[]>();
 
-    async provideCompletionItems(model: monaco.editor.ITextModel): Promise<monaco.modes.ISuggestResult> {
+    async provideCompletionItems(model: monaco.editor.ITextModel, position: monaco.Position, context: monaco.modes.SuggestContext): Promise<monaco.modes.ISuggestResult> {
         const languageId = model.getModeId(); // TODO: look up a language id at the position
         await this.loadSnippets(languageId);
-        const suggestions = this.snippets.get(languageId) || [];
+        const snippetsForLanguage = this.snippets.get(languageId) || [];
+
+        // copied and modified from https://github.com/microsoft/vscode/blob/master/src/vs/workbench/contrib/snippets/browser/snippetCompletionProvider.ts
```

Comment:
We need a CQ for it. I will help with it. Unfortunately, CQs resolution can take sometime and we cannot merge before :(

Reply:
We need to add MS copyright under ours copyright as well.

Reply:
ms copyright added